src/Polysemy/Internal/Combinators.hs

```diff
@@ -381,5 +424,151 @@ transform
 transform f (Sem m) = Sem $ \k -> m $ \u ->
   k $ hoist (transform f) $ case decomp u of
     Left g -> g
-    Right (Weaving e s wv ex ins) ->
-      injWeaving (Weaving (f e) s wv ex ins)
+    Right (Weaving e mkT lwr ex) ->
+      injWeaving (Weaving (f e) mkT lwr ex)
+
+
+-- | An effect for running monadic actions within a higher-order effect
+-- currently being interpreted.
+newtype RunH z (m :: * -> *) a where
+  RunH :: z a -> RunH z m a
```

Comment:
Why not just reuse `Embed`/`embed`? Since `z` is universally quantified in the handler passed to `interpretNew`, if the handler tries to use, say, `Embed IO`, then those uses will actually get stuck because `GHC` can't prove that `z` isn't `IO`. This can be fixed by having embedding of higher-order thunks be a completely separate effect.

Besides, I don't like overloading `Embed` that much. Having a separate effect for such a specific use-case as embedding higher-order thunks makes it easier for users to grok.

Reply:
does this mean that where you would use `liftT` you now have to `raise` or do you expect that to be mostly inferred?

Reply:
You'll need to use `raise` when working with `interpretNew` in two scenarios:

* You want to `send` a _completely polymorphic_ effect `e`, since then GHC can't tell it's not `RunH z`.
* You want to bring actions that are arguments to the interpreter into the environment.

So for example, if you want to implement `runOutputSem` using `interpretNew`:
```haskell
runOutputSem act = interpretNew \case
  Output o -> raise (act o)
```

Otherwise, you shouldn't ever need to `raise`. Indeed, most uses of `interpret` can be replaced with `interpretNew` and otherwise have the exact same definition and still work.

Reply:
awesome!